document/src/vespa/document/update/fieldpathupdate.cpp

```diff
@@ -64,14 +66,18 @@ FieldPathUpdate::applyTo(Document& doc) const
     if (_originalWhereClause.empty()) {
         doc.iterateNested(path, *handler);
     } else {
-        std::unique_ptr<select::Node> whereClause = parseDocumentSelection(_originalWhereClause, *doc.getRepo());
-        select::ResultList results = whereClause->contains(doc);
-        for (select::ResultList::const_iterator i = results.begin(); i != results.end(); ++i) {
-            LOG(spam, "vars = %s", handler->getVariables().toString().c_str());
-            if (*i->second == select::Result::True) {
-                handler->setVariables(i->first);
-                doc.iterateNested(path, *handler);
+        try {
+            std::unique_ptr<select::Node> whereClause = parseDocumentSelection(_originalWhereClause, *doc.getRepo());
+            select::ResultList results = whereClause->contains(doc);
+            for (select::ResultList::const_iterator i = results.begin(); i != results.end(); ++i) {
+                LOG(spam, "vars = %s", handler->getVariables().toString().c_str());
+                if (*i->second == select::Result::True) {
+                    handler->setVariables(i->first);
+                    doc.iterateNested(path, *handler);
+                }
             }
+        } catch (const ParsingFailedException &e) {
```

Comment:
Suggest moving try/catch and logging to parseDocumentSelection. and return a False node to get correct evaluation.